packages/core/services/DatabaseService/index.ts

```diff
@@ -7,21 +7,24 @@ import { Source } from "../../entity/FileExplorerURL";
 import SQLBuilder from "../../entity/SQLBuilder";
 import DataSourcePreparationError from "../../errors/DataSourcePreparationError";
 
-const PRE_DEFINED_COLUMNS = [
-    "File ID",
-    "File Path",
-    "File Name",
-    "File Size",
-    "Thumbnail",
-    "Uploaded",
-];
+enum PreDefinedColumn {
+    FILE_ID = "File ID",
+    FILE_PATH = "File Path",
+    FILE_NAME = "File Name",
+    FILE_SIZE = "File Size",
+    THUMBNAIL = "Thumbnail",
+    UPLOADED = "Uploaded",
+}
+const PRE_DEFINED_COLUMNS = Object.values(PreDefinedColumn);
 
 /**
  * Service reponsible for querying against a database
  */
 export default abstract class DatabaseService {
-    protected readonly SOURCE_METADATA_TABLE = "source_metadata";
     public static readonly LIST_DELIMITER = ",";
+    public static readonly HIDDEN_UID_ANNOTATION = "hidden_bff_uid";
+    // Name of the hidden column BFF uses to uniquely identify rows
```

Comment:
tiny nit: comment & line 25 might be swapped?

Reply:
Done in https://github.com/AllenInstitute/biofile-finder/pull/323/commits/99d172e35b568cbab19eb35777fcd4f302c219e3